app/services/notify_service.rb

```diff
@@ -65,7 +65,9 @@ def create_notification
   end
 
   def send_push_notifications
-    WebPushNotificationWorker.perform_async(@recipient.id, @notification.id)
+    sessions_with_subscriptions = @recipient.user.session_activations.where.not(web_push_subscription: nil)
+
+    WebPushNotificationWorker.push_bulk(sessions_with_subscriptions.pluck(:id).map { |session_activation_id| [session_activation_id, @notification.id] })
```

Comment:
Just in case you missed: you can give a block to `push_bulk` of sidekiq-bulk.
https://github.com/tootsuite/mastodon/blob/master/app/services/fan_out_on_write_service.rb#L40